resolver/resolver.go

```diff
@@ -986,6 +987,51 @@ func (r *Resolver) resolveMessageAlias(ctx *context, aliasName string) *Message
 	return r.resolveMessage(ctx, ctx.file().Package, aliasName)
 }
 
+func (r *Resolver) resolveMessageRuleValidations(ctx *context, validations []*federation.Validation) []*ValidationRule {
+	var resolved []*ValidationRule
+	for i, validation := range validations {
+		v, err := r.resolveMessageRuleValidation(i, validation)
+		if err != nil {
+			ctx.addError(
+				ErrWithLocation(
+					err.Error(),
+					source.MessageValidationLocation(ctx.fileName(), ctx.messageName(), i),
+				),
+			)
+			continue
+		}
+		resolved = append(resolved, v)
+	}
+	return resolved
+}
+
+func (r *Resolver) resolveMessageRuleValidation(idx int, validation *federation.Validation) (*ValidationRule, error) {
+	e := validation.GetError()
+
+	if e.GetRule() != "" && len(e.GetDetails()) != 0 {
+		return nil, errors.New("cannot set both rule and details at the same time")
+	}
+
+	name := validation.GetName()
+	if name == "" {
+		// the default validation name
+		name = fmt.Sprintf("MessageValidation%d", idx)
```

Comment:
Use the FQDN of the message to make it unique. e.g.) `fmt.Sprintf("%s_validation_%d", msg.FQDN(), idx)`

Reply:
Ah, reading the generated code, the FQDN did not seem to be necessary. A name ( please use lower snake case ) that is unique in the message is sufficient.